### PR TITLE
Fix 4104

### LINKED ISSUE #4104: Possible Collections collaborator pagination bug
## Expected behavior
The owner UI consistently loads for the owner of the collection.

## Current behavior
_Please include any error messages from the browser console and/or screenshots_
Current behavior as described on Talk: 
https://www.zooniverse.org/talk/17/457466?comment=764352&page=1

The summation is that sometimes the owner user is not getting the owner UI loaded consistently. Sometimes they see the navigation to edit the settings, sometimes not. 

## Steps to replicate
Possibly might be a issue with pagination on the collection roles as the reported collections with this bug have 20+ roles.
Reported collections with this issue:
https://www.zooniverse.org/collections/jrussler/75103
https://www.zooniverse.org/collections/jrussler/76283
https://www.zooniverse.org/collections/jrussler/76289

## Additional information

- **Operating system:**
- **Browser:**

### PR DESCRIPTION
**Pull request by srallen**
Fixes #4104. We weren't paginating the request for collection roles, so this adds in a callback to request for all the pages of the collection roles until we have them all.

I refactored how the collection owner is requested and pass it down as a prop to the CollectionPage instead. The include for the owner worked, but it wasn't being cached correctly into the client's `_typesCache`, so it was unretrievable (yay another client cache bug!). I removed the include request and explicitly request the for the owner user once, rather than twice which was happening with the original include with the collection request and in `componentWillMount` in the child component. 

This also adds a paginator and a bit of redesign to the collaborators editor page.

The package-lock.json has a diff because I tried out using the paginator we put in zooniverse-react-components, but changed my mind about that because I'd like to write up docs on using it first and put off including ZRC in PFE for the version 1 release (which is hopefully soon). I thought I had uncovered some bugs, but I in fact was just using the wrong type for a prop, so I want to be sure to include some docs and warnings about that. I can throw out the package-json.lock changes, but I'm guessing something updated when I did npm install again?

I've asked the volunteer who originally reported the issue to check to see if it fixes what they've reported: https://www.zooniverse.org/talk/17/457466?comment=779309

Staged at https://fix-4104.pfe-preview.zooniverse.org

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules && npm install` and app works as expected?
- [x] Did you deploy a staging branch?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [x] Are the tests passing locally and on Travis?

## Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

**Comment by srallen** [comment]
The volunteer reports back that this branch does indeed fix the issue for them: https://www.zooniverse.org/talk/17/457466?comment=779380

**Comment by eatyourgreens** [comment]
On my list to review today.

File: app/collections/collaborators.jsx [hunk 1]
```diff
@@ -285,50 +286,80 @@ export class CollectionCollaborators extends React.Component {
...
-        this.setState({ roleSets });
-      });
+  componentWillReceiveProps(nextProps) {
+    if (this.props.location.query.page !== nextProps.location.query.page) {
+      this.update(nextProps.location.query.page);
+    }
+  }
+
+  update(_page = 1) {
+    if (_page >= 1) {
```
Comment (eatyourgreens) [inline app/collections/collaborators.jsx]: Maybe `_page = parseInt(_page)` somewhere here to avoid string/number bugs.

Comment (eatyourgreens) [reply app/collections/collaborators.jsx]: Unless this is one of those cases where the API client requires numbers as strings, in which case ignore me.

**Pull request merged**